app/battles/tournament_table.py

```diff
@@ -0,0 +1,7 @@
+class Table:
+    @staticmethod
+    def results(participants: dict) -> dict:
+        return {
+            knight_name: knight.health
+            for knight_name, knight in participants.items()
+        }
```

Comment:
I think it can be done as separate piece of code inside main, no need to overload it and create class for it